### PR TITLE
Add page type configuration in page details page

### PR DESCRIPTION
**Pull request by orzechdev**
I want to merge this change because... it adds:
- page type selection on page creation
- page attribute values selection based on the selected page type
- `AttributeError` as replacement of `ProductError` in some attributes mutations

**PR intended to be tested with API branch:** 

### Screenshots
Zrzut ekranu 2020-10-30 o 13 32 53
Zrzut ekranu 2020-10-30 o 13 35 17
Zrzut ekranu 2020-10-30 o 13 32 42

### Pull Request Checklist

1. [x] All visible strings are translated with proper context.
1. [x] All data-formatting is locale-aware (dates, numbers, and so on).
1. [x] Translated strings are extracted.
1. [x] Number of API calls is optimized.
1. [x] The changes are tested.
1. [x] Data-test are added for new elements.
1. [x] Type definitions are up to date.
1. [ ] Changes are mentioned in the changelog.

### Test environment config

API_URI=https://master.staging.saleor.rocks/graphql/

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-1180 Add ability to assign page type on a page details page](https://app.clickup.com/t/2549495/SALEOR-1180)

**Comment by patrys** [comment]
Task linked: [SALEOR-1183 Handle getting and saving attribute values on a page details view](https://app.clickup.com/t/2549495/SALEOR-1183)

**Comment by patrys** [comment]
Task linked: [SALEOR-1460 Update attribute mutations permissions](https://app.clickup.com/t/2549495/SALEOR-1460)

File: src/pages/components/PageOrganizeContent/PageOrganizeContent.tsx [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+export interface PageOrganizeContentProps {
+  canChangeType: boolean;
+  data: FormData;
+  pageType?: PageTypeFragment;
+  pageTypeInputDisplayValue?: string;
+  errors: PageErrorFragment[];
+  disabled: boolean;
+  pageTypes: PageTypeFragment[];
+  onPageTypeChange?: (event: React.ChangeEvent<any>) => void;
```
Comment (dominik-zeglen) [inline src/pages/components/PageOrganizeContent/PageOrganizeContent.tsx:25]: onPageTypeChange?: FormChange;

Comment (dominik-zeglen) [inline src/pages/components/PageOrganizeContent/PageOrganizeContent.tsx:25]:
```suggestion
  onPageTypeChange?: FormChange;
```

File: src/pages/components/PageOrganizeContent/PageOrganizeContent.tsx [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+            })}
+            error={!!formErrors.pageType}
+            helperText={getPageErrorMessage(formErrors.pageType, intl)}
+            name={"pageType" as keyof FormData}
+            onChange={onPageTypeChange}
+            value={data.pageType}
+            choices={
+              pageTypes?.map(({ name, id }) => ({ label: name, value: id })) ||
+              []
+            }
```
Comment (dominik-zeglen) [inline src/pages/components/PageOrganizeContent/PageOrganizeContent.tsx:83]:
```suggestion
            choices={pageTypes ? mapNodeToChoice(pageTypes) : []}
```

File: src/pages/utils/handlers.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    const newAttributeValues = toggle(
+      value,
+      attribute.value,
+      (a, b) => a === b
+    );
+
+    triggerChange();
+    changeAttributeData(attributeId, newAttributeValues);
+  };
+}
```
Comment (dominik-zeglen) [inline src/pages/utils/handlers.ts:56]: Can we test them? They are quite complex.

**Review by dominik-zeglen — CHANGES_REQUESTED** [review]

**Review by dominik-zeglen — APPROVED** [review]

**Pull request merged**